clients/rust/tests/harvest_sync_rewards.rs

```diff
@@ -0,0 +1,905 @@
+        &mut program_test,
+        &Pubkey::new_unique(),
+        &Pubkey::new_unique(),
+    );
+    let mut context = program_test.start_with_context().await;
+    let slot = context.genesis_config().epoch_schedule.first_normal_slot + 1;
+    context.warp_to_slot(slot).unwrap();
+
+    // Given a config, validator stake and sol staker stake accounts with 1 SOL staked.
+
+    // default sync_rewards_lamports = 1_000_000 (0.001 SOL)
+    let config_manager = ConfigManager::new(&mut context).await;
+    let validator_stake_manager =
+        ValidatorStakeManager::new_with_vote(&mut context, &config_manager.config, vote).await;
+    let sol_staker_stake_manager = SolStakerStakeManager::new(
+        &mut context,
+        &config_manager.config,
+        &validator_stake_manager.stake,
+        &validator_stake_manager.vote,
+        1_000_000_000, // 1 SOL staked
+    )
+    .await;
+
+    // And there is 1 SOL for stake rewards on the config.
+
+    let mut account = get_account!(context, config_manager.config);
+    let mut config_account = Config::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the total amount delegated
+    config_account.token_amount_delegated = 1_300_000_000;
+    config_account.accumulated_stake_rewards_per_token =
+        calculate_stake_rewards_per_token(1_300_000_000, 1_300_000_000);
+
+    account.lamports += 1_300_000_000; // 1 SOL
+    account.data = config_account.try_to_vec().unwrap();
+    context.set_account(&config_manager.config, &account.into());
+
+    // And the SOL staker stake has 1_300_000_000 tokens staked.
+
+    let mut account = get_account!(context, sol_staker_stake_manager.stake);
+    let mut stake_account = SolStakerStake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the staked values:
+    //   - delegation amount = 1_300_000_000
+    //   - lamports amount = 1_000_000_000
+    stake_account.delegation.amount = 1_300_000_000;
+    stake_account.lamports_amount = 1_000_000_000;
+    account.data = stake_account.try_to_vec().unwrap();
+    context.set_account(&sol_staker_stake_manager.stake, &account.into());
+
+    // And the SOL staker stake and validator stake accounts are correctly synced.
+
+    let account = get_account!(context, sol_staker_stake_manager.stake);
+    let stake_account = SolStakerStake::from_bytes(account.data.as_ref()).unwrap();
+
+    assert_eq!(stake_account.lamports_amount, 1_000_000_000);
+
+    let account = get_account!(context, validator_stake_manager.stake);
+    let validator_stake_account = ValidatorStake::from_bytes(account.data.as_ref()).unwrap();
+    assert_eq!(
+        validator_stake_account.total_staked_lamports_amount,
+        1_000_000_000
+    );
+
+    // And we close the SOL stake account.
+
+    context.set_account(
+        &stake_account.sol_stake,
+        &AccountSharedData::from(Account {
+            data: vec![],
+            lamports: 0,
+            owner: system_program::ID,
```

Comment:
You might want to check this once locally, but it's possible withdraw everything from a stake account and then use it again within the same transaction because the stake program doesn't reallocate to 0 / set the owner to the system program.

So what happens if the owner is the stake program and there's no data?

Reply:
Works the same way. Should I keep `system_program::ID` or the stake program on the test, or both?

Reply:
Ok great! No worries, feel free to keep this